### PR TITLE
test(idempotency): migrate to vitest

### LINKED ISSUE #3123: Maintenance: migrate test runner to vitest for idempotency package
### Summary

To continue the migration from Jest to Vitest we need to move over the tests for the idempotency package to use the new test runner.

### Why is this needed?

So that we can continue our efforts of removing dev dependencies and streamline the tests.

### Which area does this relate to?

Idempotency

### Solution

_No response_

### Acknowledgment

- [X] This request meets [Powertools for AWS Lambda (TypeScript) Tenets](https://docs.powertools.aws.dev/lambda/typescript/latest/#tenets)
- [ ] Should this be considered in other Powertools for AWS Lambda languages? i.e. [Python](https://github.com/aws-powertools/powertools-lambda-python/), [Java](https://github.com/aws-powertools/powertools-lambda-java/), and [.NET](https://github.com/aws-powertools/powertools-lambda-dotnet/)

### Future readers

Please react with 👍 and your use case to help us understand customer demand.

### PR DESCRIPTION
**Pull request by dreamorosi**
### Changes

> Please provide a summary of what's being changed

This PR updates the unit tests for the Idempoency package so that they use vitest instead of Jest.

The e2e tests for the utility will be addressed in a future PR. I also removed a number of test cases that were redundant while leaving coverage unaffected. There's still room for further improvement but I decided to not address it now for the sake of keeping the PR focused.

> Please add the issue number below, if no issue is present the PR might get blocked and not be reviewed

**Issue number:** closes #3123

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Disclaimer**: We value your time and bandwidth. As such, any pull requests created on non-triaged issues might not be successful.

### CODE REVIEW

File: packages/idempotency/tests/unit/IdempotencyHandler.test.ts [hunk 1]
```diff
@@ -115,7 +109,7 @@ describe('Class IdempotencyHandler', () => {
       expect(mockResponseHook).not.toHaveBeenCalled();
     });
 
-    test('when response hook is provided, it should should call responseHook during an idempotent request', () => {
+    /* it('when response hook is provided, it should should call responseHook during an idempotent request', () => {
```
Comment (leandrodamascena) [inline packages/idempotency/tests/unit/IdempotencyHandler.test.ts:112]: Is this intentional or do you need to uncomment?

**Review by leandrodamascena — DISMISSED** [review]
Hey @dreamorosi! I see that you basically moved things from jest to vitest and reorganized files. If CI is green I'm good with this PR.

I just left a comment to check if the comment was intentional or not.
APPROVED!

**Review by am29d — APPROVED** [review]
Nice! 👌

File: packages/idempotency/tests/unit/IdempotencyHandler.test.ts (hunk 1 above)
Comment (dreamorosi) [reply packages/idempotency/tests/unit/IdempotencyHandler.test.ts:112]: Should have removed this

**Comment by dreamorosi** [comment]
Sorry, I had to make another commit to remove a leftover test I had left commented out.

@am29d, @leandrodamascena, if you could click approve again please.

**Review by leandrodamascena — APPROVED** [review]

**Pull request merged**